src/vm/hints/secp/field_utils.rs

```diff
@@ -78,36 +74,122 @@ pub fn reduce(
     let x_d1 = get_integer_from_relocatable_plus_offset(&x_reloc, 1, vm)?;
     let x_d2 = get_integer_from_relocatable_plus_offset(&x_reloc, 2, vm)?;
 
-    //SECP_P = 2**256 - 2**32 - 2**9 - 2**8 - 2**7 - 2**6 - 2**4 - 1
-    let sec_p = bigint_str!(
-        b"115792089237316195423570985008687907853269984665640564039457584007908834671663"
-    );
-
-    let value = pack(x_d0, x_d1, x_d2, &vm.prime).mod_floor(&sec_p);
+    let value = pack(x_d0, x_d1, x_d2, &vm.prime).mod_floor(&SECP_P);
 
     vm.exec_scopes
         .assign_or_update_variable("value", PyValueType::BigInt(value));
 
     Ok(())
 }
 
+/*
+Implements hint:
+%{
+    from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+
+    x = pack(ids.x, PRIME) % SECP_P
+%}
+*/
+pub fn is_zero_pack(
+    vm: &mut VirtualMachine,
+    ids: &HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    let x_reloc = get_relocatable_from_var_name("x", ids, vm, hint_ap_tracking)?;
+
+    let x_d0 = get_integer_from_relocatable_plus_offset(&x_reloc, 0, vm)?;
+    let x_d1 = get_integer_from_relocatable_plus_offset(&x_reloc, 1, vm)?;
+    let x_d2 = get_integer_from_relocatable_plus_offset(&x_reloc, 2, vm)?;
+
+    let x = (pack(x_d0, x_d1, x_d2, &vm.prime)).mod_floor(&SECP_P);
+
+    vm.exec_scopes
+        .assign_or_update_variable("x", PyValueType::BigInt(x));
+
+    Ok(())
+}
+/*
+Implements hint:
+in .cairo program
+if nondet %{ x == 0 %} != 0:
+
+On .json compiled program
+"memory[ap] = to_felt_or_relocatable(x == 0)"
+*/
+pub fn is_zero_nondet(vm: &mut VirtualMachine) -> Result<(), VirtualMachineError> {
+    // get `value` variable from vm scope
+    let x: &BigInt = match vm
+        .exec_scopes
+        .get_local_variables()
+        .ok_or(VirtualMachineError::ScopeError)?
+        .get("x")
+    {
+        Some(PyValueType::BigInt(x)) => x,
+        _ => {
+            return Err(VirtualMachineError::VariableNotInScopeError(String::from(
+                "x",
+            )))
+        }
+    };
```

Comment:
I think one of the helpers @fmoletta added can make this more brief.

Reply:
Nice, this will simplify things even more. 
Done